### ISSUE TITLE
[PrestoS3FileSystem] Cannot do a CTAS in an empty database

### ISSUE THREAD

**Issue by nezihyigitbasi** [prestodb/presto#4482]
If I create a new database and then do a CTAS to create a new table under it, it fails with exception below. I believe the problem is [here](https://github.com/facebook/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java#L437-L440), because with a fresh database created through the Hive version we use (Emr Hive 0.14) the corresponding object created on S3 will be `s3n://path/nezih_test.db_$folder$` (I believe `NativeS3FileSystem` creates directories with this suffix) and this part of the code assumes that it will be `s3n://path/nezih_test.db/`. 

```Java
presto:nezih_test> create table x as select 1 as x;
Query 20160205_000550_00004_ijvji failed: Database 'nezih_test' location does not exist: s3n://path/nezih_test.db
com.facebook.presto.spi.PrestoException: Database 'nezih_test' location does not exist: s3n://path/nezih_test.db
	at com.facebook.presto.hive.HiveWriteUtils.getTableDefaultLocation(HiveWriteUtils.java:321)
	at com.facebook.presto.hive.HiveLocationService.forNewTable(HiveLocationService.java:51)
	at com.facebook.presto.hive.HiveMetadata.beginCreateTable(HiveMetadata.java:612)
	at com.facebook.presto.hive.HiveMetadata.beginCreateTable(HiveMetadata.java:126)
	at com.facebook.presto.spi.classloader.ClassLoaderSafeConnectorMetadata.beginCreateTable(ClassLoaderSafeConnectorMetadata.java:190)
	at com.facebook.presto.transaction.LegacyConnectorMetadata.beginCreateTable(LegacyConnectorMetadata.java:167)
	at com.facebook.presto.metadata.MetadataManager.beginCreateTable(MetadataManager.java:532)
	at com.facebook.presto.sql.planner.optimizations.BeginTableWrite$Rewriter.createWriterTarget(BeginTableWrite.java:150)
	at com.facebook.presto.sql.planner.optimizations.BeginTableWrite$Rewriter.visitTableFinish(BeginTableWrite.java:120)
	at com.facebook.presto.sql.planner.optimizations.BeginTableWrite$Rewriter.visitTableFinish(BeginTableWrite.java:74)
	at com.facebook.presto.sql.planner.plan.TableFinishNode.accept(TableFinishNode.java:80)
	at com.facebook.presto.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:86)
	at com.facebook.presto.sql.planner.plan.SimplePlanRewriter$RewriteContext.lambda$defaultRewrite$145(SimplePlanRewriter.java:75)
	at com.facebook.presto.sql.planner.plan.SimplePlanRewriter$RewriteContext$$Lambda$312/1028397322.apply(Unknown Source)
```

**Comment by nezihyigitbasi** [comment]
We cannot simply get rid of the appended `\` character to the key as it may result in lots of objects being returned in the worst case (objects with the same prefix). One possible (inefficient) solution is to make a second list call, but this time with `_$folder$` appended.

**Comment by tomz** [comment]
This has to do with the following lines in HiveWriteUtils.java:

```
        Path databasePath = new Path(location);
        if (!pathExists(hdfsEnvironment, databasePath)) {
            throw new PrestoException(HIVE_DATABASE_LOCATION_ERROR, format("Database '%s' location does not exist: %s", schemaName, databasePath));
        }
        if (!isDirectory(hdfsEnvironment, databasePath)) {
            throw new PrestoException(HIVE_DATABASE_LOCATION_ERROR, format("Database '%s' location is not a directory: %s", schemaName, databasePath));
        }

```
The following fixes the issue, but I'd like to know if there's a better way because submitting a PR:

```
        Path databasePath = new Path(location);
        if (!pathExists(hdfsEnvironment, databasePath) && !location.startsWith("s3")) {
            throw new PrestoException(HIVE_DATABASE_LOCATION_ERROR, format("Database '%s' location does not exist: %s", schemaName, databasePath));
        }
        if (!isDirectory(hdfsEnvironment, databasePath) && !location.startsWith("s3")) {
            throw new PrestoException(HIVE_DATABASE_LOCATION_ERROR, format("Database '%s' location is not a directory: %s", schemaName, databasePath));
        }
```

**Comment by electrum** [comment]
Fixed in #5246
